drivers/include/nrf24l01p.h

```diff
@@ -1,627 +1,840 @@
+ * @brief   Struct that holds all active configuration values
  */
 typedef struct {
-    spi_t spi;           /**< SPI device to initialize */
-    gpio_t ce;           /**< GPIO pin to initialize as chip enable */
-    gpio_t cs;           /**< GPIO pin to initialize as chip select */
-    gpio_t irq;          /**< GPIO pin to initialize as interrupt request */
-    unsigned listener;   /**< Place to store an ID in */
+    /**
+     * @brief Padding for byte alignment
+     */
+    uint8_t :                   5;
+    /**
+     * @brief   Currently configured protocol (Do not change after initialization)
+     *          @see nrf24l01p_protocol_t
+     */
+    uint8_t cfg_protocol :       1;
+    /**
+     * @brief   Current CRC length configuration value
+     *          @see nrf24l01p_crc_t
+     */
+    uint8_t cfg_crc :            2;
+    /**
+     * @brief   Current tx power configuration value
+     *          @see nrf24l01p_tx_power_t
+     */
+    uint8_t cfg_tx_power :       2;
+    /**
+     * @brief   Current data rate configuration value
+     *          @see nrf24l01p_rfdr_t
+     */
+    uint8_t cfg_data_rate :      2;
+    /**
+     * @brief   Current channel
+     */
+    uint8_t cfg_channel :        4;
+    /**
+     * @brief   Current address width configuration value (Do not change after initialization
+     *          @see nrf24l01p_aw_t
+     */
+    uint8_t cfg_addr_width :     2;
+    /**
+     * @brief   Current maximum number of retransmissions
+     *          (Only used if protocol is ESB)
+     */
+    uint8_t cfg_max_retr :       4;
+    /**
+     * @brief   Current retransmission delay configuration value
+     *          (Only used if protocol is ESB) @see nrf24l01p_ard_t
+     */
+    uint8_t cfg_retr_delay :     4;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 0 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p0 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 1 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p1 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 2 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p2 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 3 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p3 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 4 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p4 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 5 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p5 :    5;
+} nrf24l01p_cfg_t;
```

Comment:
You are right.
Without `__attribute__((packed))` it is 10 bytes large.
With `__attribute__((packed))` it is 7 bytes large.
I thought it would be aligned to 1 Byte because no member is larger than 8 bit.